### ISSUE TITLE
Object has no attribute 'registers' error on register types 1 and 2

### ISSUE THREAD

**Issue by cybcon** [cybcon/modbus-client#17]
While analyzing issue on modbus-server (https://github.com/cybcon/modbus-server/issues/18) I face another issue with the modbus-client when trying to get values of register type 1 (Discrete Output Coils) and 2 (Discrete Input Contacts):

Error is ´object has no attribute 'registers'`:
```
# docker run --rm oitc/modbus-client:latest -s 192.168.57.10 -p 5020 -t 1 -r 0 -l 10
Traceback (most recent call last):
  File "/app/modbus_client.py", line 177, in <module>
    df = parse_modbus_result(rr.registers, register_number, big_endian=args.bigEndian)
AttributeError: 'ReadCoilsResponse' object has no attribute 'registers'


# docker run --rm oitc/modbus-client:latest -s 192.168.57.10 -p 5020 -t 2 -r 0 -l 10
Traceback (most recent call last):
  File "/app/modbus_client.py", line 177, in <module>
    df = parse_modbus_result(rr.registers, register_number, big_endian=args.bigEndian)
AttributeError: 'ReadDiscreteInputsResponse' object has no attribute 'registers'
```

**Comment by cybcon** [comment]
Modbus-server is configured with:

```
[..]
"zeroMode": true,
"initializeUndefinedRegisters": true,
"discreteInput": {},
"coils": {},
[..]
```

and the modbus-server log shows following output during that tests:
```
2023-11-05 10:50:26,945 MainThread      DEBUG    sync           :348      Started thread to serve client at ('10.243.0.1', 58906)
2023-11-05 10:50:26,947 Thread-3 (process_request_thread) DEBUG    sync           :46       Client Connected [10.243.0.1:58906]
2023-11-05 10:50:26,947 Thread-3 (process_request_thread) DEBUG    sync           :199      Handling data: 0x0 0x1 0x0 0x0 0x0 0x6 0x1 0x1 0x0 0x0 0x0 0xa
2023-11-05 10:50:26,948 Thread-3 (process_request_thread) DEBUG    socket_framer  :147      Processing: 0x0 0x1 0x0 0x0 0x0 0x6 0x1 0x1 0x0 0x0 0x0 0xa
2023-11-05 10:50:26,948 Thread-3 (process_request_thread) DEBUG    factory        :137      Factory Request[ReadCoilsRequest: 1]
2023-11-05 10:50:26,949 Thread-3 (process_request_thread) DEBUG    context        :63       validate: fc-[1] address-0: count-10
2023-11-05 10:50:26,950 Thread-3 (process_request_thread) DEBUG    context        :77       getValues fc-[1] address-0: count-10
2023-11-05 10:50:26,950 Thread-3 (process_request_thread) DEBUG    sync           :229      send: [ReadCoilsResponse(10)]- b'0001000000050101020000'
2023-11-05 10:50:26,956 Thread-3 (process_request_thread) DEBUG    sync           :199      Handling data:
2023-11-05 10:50:26,957 Thread-3 (process_request_thread) DEBUG    socket_framer  :147      Processing:
2023-11-05 10:50:26,957 Thread-3 (process_request_thread) DEBUG    sync           :54       Client Disconnected [10.243.0.1:58906]
2023-11-05 10:50:33,840 MainThread      DEBUG    sync           :348      Started thread to serve client at ('10.243.0.1', 58922)
2023-11-05 10:50:33,841 Thread-4 (process_request_thread) DEBUG    sync           :46       Client Connected [10.243.0.1:58922]
2023-11-05 10:50:33,842 Thread-4 (process_request_thread) DEBUG    sync           :199      Handling data: 0x0 0x1 0x0 0x0 0x0 0x6 0x1 0x2 0x0 0x0 0x0 0xa
2023-11-05 10:50:33,843 Thread-4 (process_request_thread) DEBUG    socket_framer  :147      Processing: 0x0 0x1 0x0 0x0 0x0 0x6 0x1 0x2 0x0 0x0 0x0 0xa
2023-11-05 10:50:33,843 Thread-4 (process_request_thread) DEBUG    factory        :137      Factory Request[ReadDiscreteInputsRequest: 2]
2023-11-05 10:50:33,843 Thread-4 (process_request_thread) DEBUG    context        :63       validate: fc-[2] address-0: count-10
2023-11-05 10:50:33,844 Thread-4 (process_request_thread) DEBUG    context        :77       getValues fc-[2] address-0: count-10
2023-11-05 10:50:33,844 Thread-4 (process_request_thread) DEBUG    sync           :229      send: [ReadDiscreteInputsResponse(10)]- b'0001000000050102020000'
2023-11-05 10:50:33,852 Thread-4 (process_request_thread) DEBUG    sync           :199      Handling data:
2023-11-05 10:50:33,856 Thread-4 (process_request_thread) DEBUG    socket_framer  :147      Processing:
2023-11-05 10:50:33,859 Thread-4 (process_request_thread) DEBUG    sync           :54       Client Disconnected [10.243.0.1:58922]
```

**Comment by cybcon** [comment]
Related issue in modbus-.server: https://github.com/cybcon/modbus-server/issues/19
Coils and discrete input registers are boolean (bit) variables and not words (2 byte).
Handling of register types 1 and 2 needs to be adjusted.

**Comment by cybcon** [comment]
Fixed with PR #18
